assets/app/views/editor/edit-file.js

```diff
@@ -143,13 +153,43 @@ var EditorView = Backbone.View.extend({
     if (settings) this.doc.frontMatter = settings;
     if (content) this.doc.content = content;
 
-    this.model.commit({
-      content: this.doc.toMarkdown(),
-      message: this.$('#save-content-message').val()
-    });
+    if (this.isNewPage) {
+      pageTitle = fileNameFromTitle(yaml.parse(settings)['title']);
+      pageTitle = pageTitle.toLowerCase().replace(/\s/g, '-');
```

Comment:
Should we tighten this to something like `/^\w/`, which should filter for only alphanumeric and _ characters